cf-agent/cf-agent.c

```diff
@@ -796,8 +798,12 @@ static void ThisAgentInit(void)
 */
 
     snprintf(filename, CF_BUFSIZE, "%s/cfagent.%s.log", GetLogDir(), VSYSNAME.nodename);
+    ToLowerStrInplace(filename);
```

Comment:
I don't quite understand what these `ToLower` calls are doing in this commit which is supposed to just make sure the permissions are correct. And shouldn't this be done in `MapName()` if we need it?

Reply:
The policy that manages this log file assumes lower case but the code did not. See https://github.com/cfengine/masterfiles/blob/master/inventory/any.cf#L204 for one example. @nickanderson I think you mentioned this to me. Is this really a problem? If so, I will move this change to another ticket/PR. :)

Reply:
Thanks for the explanation. Let's see what @nickanderson says, but it should definitely be part of that particular commit.

Reply:
From some quick research, it seems that when this policy was introduced the understanding was the log file was created from hostname -f and that command always returns lowercase.

```
When the agent observes a file that is setuid it is logged to $(sys.workdir)/cfagent.$(hostname -f).log.

This change introduces inventory for these observed files. For each file in the log that is setuid inventory of that file is tagged with setuid Files. Additionally separate inventory is done for each root owned setuid file as root owned setuid Files. This policy also handles clearing of files that were once setuid but are no longer found or are no longer setuid from the log since there is no pre-existing automatic processes to clear entries from the log.
```

So the policy looks for candidates that have a matching alll lowercase form. If this ToLower makes sure that the log file name is lowercase then it makes sense to me. Perhaps we were relying on an unenforced expectation before.

```
      "setuid_log_path"
        comment => "We select the file that matches the downcased version of the
                    hostname since sys.fqhost always returns lower case",
        string => "$(candidates)",
        if => strcmp( "$(sys.workdir)/cfagent.$(sys.fqhost).log",
string_downcase($(candidates)));
```